Public/Src/Tools/ContentPlacementAnalysisTools/ContentPlacementAnalysisTools.Extraction/Action/BuildDownload.cs

```diff
@@ -0,0 +1,147 @@
+        private string m_bxlPath = null;
+        private string m_dominoPath = null;
+        private string m_outputDirectory = null;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public BuildDownload(ApplicationConfiguration config, string outputDirectory)
+        {
+            m_configuration = config;
+            m_outputDirectory = outputDirectory;
+        }
+
+        /// <inheritdoc />
+        protected override void CleanUp(KustoBuild input, BuildDownloadOutput output) {}
+
+        /// <inheritdoc />
+        protected override void Setup(KustoBuild input)
+        {
+            // do some checks in here
+            m_bxlPath = Path.Combine(m_outputDirectory, input.BuildControllerMachineName);
+            m_dominoPath = Path.Combine(m_outputDirectory, input.BuildControllerMachineName, "Domino");
+            // create the directories
+            Directory.CreateDirectory(m_bxlPath);
+            Directory.CreateDirectory(m_dominoPath);
+            Contract.Requires(Directory.Exists(m_bxlPath) && Directory.Exists(m_dominoPath), "Could not create directories for downloads...");
+        }
+
+        /// <summary>
+        /// Given a machine and log dir, downloads the zip files used to reconstructuc the build log 
+        /// </summary>
+        protected override BuildDownloadOutput Perform(KustoBuild input)
+        {
+            s_logger.Debug($"BuildDownloader starts...");
+            try
+            {
+                var bxlZip = Path.Combine(m_bxlPath, "BuildXLLogs.zip");
+                var dominoZip = Path.Combine(m_dominoPath, "Domino.zip");
+                // build the urls
+                var urls = BuildDownloadUrls(input);
+                var downloaded = 0;
+                // and download the two of them
+                foreach (var url in urls)
+                {
+                    // download each one
+                    if (DownloadFileTo(url, bxlZip, dominoZip))
+                    {
+                        ++downloaded;
+                    }
+                }
+                // now, the output contains the number of downloaded files (that should be two)
+                // and the base directory of where they are
+                return new BuildDownloadOutput(m_bxlPath, downloaded, bxlZip, dominoZip, input);
+            }
+            finally
+            {
+                s_logger.Debug($"BuildDownloader ends in {Stopwatch.ElapsedMilliseconds}ms");
+            }
+            
+        }
+
+        private string[] BuildDownloadUrls(KustoBuild buildData) => new string[] {
+            $"https://b/getfile?path=\\\\{buildData.BuildControllerMachineName}\\{buildData.LogDirectory}\\Logs\\ProductBuild\\BuildXLLogs.zip",
+            $"https://b/getfile?path=\\\\{buildData.BuildControllerMachineName}\\{buildData.LogDirectory}\\Logs\\ProductBuild\\Domino\\Domino.zip"
+        };
+
+        private bool DownloadFileTo(string url, string bxl, string domino)

```

Comment:
Knowing batmon, we might run into download limmitation for large files. Fine for now, but this should probably be hardnened in a retry loop...

Ideally this will use the Logs Drop feature that is being built in CloudBuild.